src/shared/logs-show.c

```diff
@@ -403,24 +404,27 @@ static int output_timestamp_realtime(
                         break;
 
                 case OUTPUT_SHORT_ISO:
-                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S%z",
+                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S",
                                      localtime_or_gmtime_r(&t, &tm, flags & OUTPUT_UTC)) <= 0)
                                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                        "Failed to format ISO time");
-                        break;
-
-                case OUTPUT_SHORT_ISO_PRECISE: {
-                        char usec[7];
+                        timezone_off = 19;
+                        goto colonised_timezone;
 
+                case OUTPUT_SHORT_ISO_PRECISE:
                         /* No usec in strftime, so we leave space and copy over */
-                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S.xxxxxx%z",
+                        if (strftime(buf, sizeof(buf), "%Y-%m-%dT%H:%M:%S.",
                                      localtime_or_gmtime_r(&t, &tm, flags & OUTPUT_UTC)) <= 0)
                                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                                        "Failed to format ISO-precise time");
-                        xsprintf(usec, "%06"PRI_USEC, display_ts->realtime % USEC_PER_SEC);
-                        memcpy(buf + 20, usec, 6);
-                        break;
-                }
+                        snprintf(buf + 20, ELEMENTSOF(buf) - 20, "%06"PRI_USEC, display_ts->realtime % USEC_PER_SEC);
+                        timezone_off = 26;
+                        goto colonised_timezone;
+                colonised_timezone: {
```

Comment:
uh?

Reply:
yep. definitely wasn't happy with how i wrote it the first time. should be less insane now